inspire-atom/src/main/java/org/fao/geonet/services/inspireatom/AtomPredefinedFeed.java

```diff
@@ -0,0 +1,291 @@
+import org.fao.geonet.inspireatom.util.InspireAtomUtil;
+import org.fao.geonet.kernel.DataManager;
+import org.fao.geonet.kernel.search.MetaSearcher;
+import org.fao.geonet.kernel.search.SearchManager;
+import org.fao.geonet.kernel.search.SearcherType;
+import org.fao.geonet.kernel.setting.SettingManager;
+import org.fao.geonet.kernel.setting.Settings;
+import org.fao.geonet.lib.Lib;
+import org.fao.geonet.repository.MetadataRepository;
+import org.fao.geonet.utils.Log;
+import org.fao.geonet.utils.Xml;
+
+import org.jdom.Document;
+import org.jdom.Element;
+import org.jdom.Text;
+import org.jdom.xpath.XPath;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.request.NativeWebRequest;
+
+
+/**
+ * Services for converting Service and Data metadata into INSPIRE ATOM feeds
+ *
+ * @author fgravin on 7/15/15.
+ * @author Emanuele Tajariol (etj at geo-solutions dot it)
+ */
+
+@Controller
+public class AtomPredefinedFeed {
+
+    /**
+     * Main entry point for local service ATOM feed description
+     *
+     * @param uiLang the language parameter
+     * @param uuid identifier of the metadata of service (this could be made optional once a system-wide top level metadata could be set)
+     */
+    @RequestMapping(value = "/{uiLang}/atom.predefined.service")
+    @ResponseBody
+    public HttpEntity<byte[]> localServiceDescribe(
+            @PathVariable String uiLang,
+            @RequestParam("uuid") String uuid,
+            NativeWebRequest webRequest) throws Exception {
+
+        ServiceContext context = createServiceContext(uiLang, webRequest.getNativeRequest(HttpServletRequest.class));
+
+        SettingManager sm = context.getBean(SettingManager.class);
+        boolean inspireEnable = sm.getValueAsBool(Settings.SYSTEM_INSPIRE_ENABLE);
+        if (!inspireEnable) {
+            Log.info(Geonet.ATOM, "INSPIRE is disabled");
+            throw new OperationNotAllowedEx("INSPIRE option is not enabled on this catalog.");
+        }
+
+        Element feed = getServiceFeed(context, uuid);
+        return writeOutResponse(Xml.getString(feed));
+    }
+
+    /**
+     * Main entry point for local dataset ATOM feed description.
+     *
+     * @param uiLang the language parameter
+     * @param spIdentifier the spatial dataset identifier
+     * @param spNamespace the spatial dataset namespace (not used for the moment)
+     * @param webRequest the request object
+     * @return
+     * @throws Exception
+     */
+    @RequestMapping(value = "/{uiLang}/atom.predefined.dataset")
```

Comment:
Would it make sense to add these new services in the API and not as old Jeeves path way - that gave them more visibility in the API doc - but maybe it is not relevant ?